ecosystem/typescript/sdk/src/transactions/tests/account_sequence_number.test.ts

```diff
@@ -0,0 +1,89 @@
+import { AptosAccount } from "../../account";
+import { Provider } from "../../providers";
+import { AccountSequenceNumber } from "../account_sequence_number";
+import { getFaucetClient, longTestTimeout, PROVIDER_LOCAL_NETWORK_CONFIG } from "../../tests/unit/test_helper.test";
+
+const provider = new Provider(PROVIDER_LOCAL_NETWORK_CONFIG);
+const account = new AptosAccount();
+const faucet = getFaucetClient();
+const accountSequenceNumber = new AccountSequenceNumber(provider, account);
+let getAccountSpy: jest.SpyInstance;
+
+let lastSeqNumber: bigint | null;
+
+describe("account sequence number", () => {
+  beforeAll(async () => {
+    await faucet.fundAccount(account.address(), 1000000);
+  }, longTestTimeout);
+
+  beforeEach(() => {
+    getAccountSpy = jest.spyOn(provider, "getAccount");
+  });
+
+  afterEach(() => {
+    getAccountSpy.mockRestore();
+  });
+
+  it("initializes with correct sequence number", async () => {
+    await accountSequenceNumber.initialize();
+    expect(accountSequenceNumber.currentNumber).toEqual(BigInt(0));
+    expect(accountSequenceNumber.lastUncommintedNumber).toEqual(BigInt(0));
+  });
+
+  it("updates with correct sequence number", async () => {
+    getAccountSpy.mockResolvedValue({
+      sequence_number: "2",
```

Comment:
nit: Can assign val 2 to a var, then we can reuse it on `.toEqual(BigInt(seq_num));`